File: android/src/androidTest/java/org/ligi/gobandroidhd/helper/CellFactory.java
package org.ligi.gobandroidhd.helper;

import java.util.ArrayList;
import java.util.List;
import org.ligi.gobandroid_hd.logic.Cell;
import org.ligi.gobandroid_hd.logic.CellImpl;

/**
 * The Petri dish
 */
public class CellFactory {

    public static List<Cell> getAllCellsForRect(final int sizeX, final int sizeY) {
        final List<Cell> created = new ArrayList<>();

        for (int y = 0; y < sizeY; y++) {
            for (int x = 0; x < sizeX; x++) {
                created.add(new CellImpl(x, y));
            }
        }
        return created;
    }

}

File: android/src/androidTest/java/org/ligi/gobandroid_hd/test_helper_functions/CellFactory.kt
package org.ligi.gobandroid_hd.test_helper_functions

import org.ligi.gobandroid_hd.logic.CellImpl

/**
 * The Petri dish
 */
fun getAllCellsForRect(sizeX: Int, sizeY: Int)
        = (0..sizeX * sizeY - 1).map { CellImpl(it % sizeY, it / sizeY) }
